rocky/rocky/templates/tasks/partials/task_filter.html

```diff
@@ -1,20 +1,46 @@
 {% load i18n %}
+<div class="filter">
+  <div>
+      <p>{% translate 'Filter' %}</p>
+      <button aria-expanded="false"
+              data-hide-filters-label="{% translate 'Hide filters' %}">
+        {% translate 'Show filters' %}
+      </button>
+  </div>
 
-{% if task_filter %}
-  <form method="get">
-    {% csrf_token %}
-    <label for="selected-tasks">{% translate 'Number of tasks showed' %}</label>
-    <select id="selected-tasks" name="selected-tasks">
-      {% for task in task_filter %}
-        <option {% if task == show_tasks %}selected{% endif %} value={{ task }}>
-          {% if task == "*" %}
-            {% translate 'Show all' %}
-          {% else %}
-            {{ task }}
+  <form action="" method="get" novalidate method="get">
```

Comment:
the action can be removed if empty, it will just post to the current url.
You also set the method to get twice.
Novalidate is also not required as no fields have any patterns or requirements.